### PR TITLE
fixed shap summary plot bug for binary classification models

### PR DESCRIPTION
**Pull request by pycaret**
## Related Issuse or bug
  - #1312 

Fixes: #[issue number that will be closed through this PR] #1312

#### Describe the changes you've made
A clear and concise description of what you have done to successfully close your assigned issue. Any new files? or anything you feel to let us know!

Currently, when the target variable is binary the 'summary' plot from the interpret_model shows a bar as opposed to beeswarm.

image

image

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
A clear and concise description of it.

## Checklist:

- [X] My code follows the style guidelines of this project.
- [X] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Screenshots

 Original           | Updated
 :--------------------: |:--------------------:
 **original screenshot**  | <b>updated screenshot </b> |

### CODE REVIEW

File: pycaret/internal/tabular.py [hunk 1]
```diff
@@ -7476,7 +7476,15 @@ def summary(show: bool = True):
...
-        shap_plot = shap.summary_plot(shap_values, test_X, show=show, **kwargs)
+
+        if len(shap_values) == 2:
+            try:
+                shap_plot = shap.summary_plot(shap_values[1], test_X, show=show, **kwargs)
+            except:
+                shap_plot = shap.summary_plot(shap_values, test_X, show=show, **kwargs)
+        else:
+            shap_plot = shap.summary_plot(shap_values, test_X, show=show, **kwargs)
+
```
Comment (Yard1) [inline pycaret/internal/tabular.py:7487]:
```suggestion
        try:
        	assert len(shap_values) == 2
            shap_plot = shap.summary_plot(shap_values[1], test_X, show=show, **kwargs)
        except Exception:
            shap_plot = shap.summary_plot(shap_values, test_X, show=show, **kwargs)
        
```

**Review by Yard1 — CHANGES_REQUESTED** [review]
I'd suggest this for more concise code. Also using bare `except` is not recommended, do `except Exception`

File: pycaret/internal/tabular.py (hunk 1 above)
Comment (pycaret) [reply pycaret/internal/tabular.py:7487]: Thanks - Will this still check for len(shap_values) == 2 before falling back on exception?

Comment (Yard1) [reply pycaret/internal/tabular.py:7487]: It will have the exact same logic as the code before. If the assertion is not met it will do the `except` block.

Comment (pycaret) [reply pycaret/internal/tabular.py:7487]: Great thanks

**Review by Yard1 — APPROVED** [review]

**Pull request merged**